_posts/2018-01-05-python101.md

```diff
@@ -0,0 +1,734 @@
+    List of Courses Teaching
+  </td>
+  <td>
+    Change department <br>
+    Change position <br>
+    Add course taught <br>
+    Remove course taught
+  </td>
+</tr>
+</table>
+
+Having written down the properties and actions we associate with each object, we can begin writings classes for each object. What’s a class? It’s simply a blueprint for what properties and actions an object should have. In other words, I have a class that defines a generic object, like Student. When I want specific students, such as George or Michael, I create an instance of Student, with each student’s name property set appropriately. The code might help clarify things a bit.
+
+```python
+class Student:
+  def __init__(self, first, mi, last):
+    self.first_name = first
+    self.middle_initial = mi
+    self.last_name = last
+    self.enrolled_courses = []
+
+  def enroll(self, course):
+    self.enrolled_courses.append(course)
+    course.enrolled_students.append(self)
+
+  def drop(self, course):
+    self.enrolled_courses.remove(course)
+    course.enrolled_students.remove(self)
+
+  def get_transcript(self):
+    print(self.first_name + " " + self.middle_initial + ". " + self.last_name + ":")
+    for course in self.enrolled_courses:
+      print("\t" + course.course_name + ": " + str(course.get_overall_grade_for(self)))
+
+
+class Course:
+  def __init__(self, name):
+    self.course_name = name
+    self.enrolled_students = []
+    self.assignments = {}
+
+  def add_assignment(self, assignment):
+    self.assignments[assignment] = {}
+
+  def remove_assignment(self, assignment):
+    del self.assignments[assignment]
+
+  def set_grade(self, assignment, student, grade):
+    self.assignments[assignment][student] = grade
+
+  def get_overall_grade_for(self, student):
+    sum_scores = 0
+    num_graded = 0
+    for assignment, grades in self.assignments.items():
+      if student in grades.keys():
+        sum_scores += grades[student]
+        num_graded += 1
+
+    if num_graded == 0:
+      return "N/A"
+
+    return sum_scores / float(num_graded)
+
+
+john = Student("John", "A", "Smith")
+katie = Student("Katie", "M", "Zhu")
+
+biology = Course("Biology")
+calculus = Course("Calculus")
+physics = Course("Physics")
+
+john.enroll(biology)
+john.enroll(calculus)
+
+katie.enroll(biology)
+katie.enroll(calculus)
+katie.enroll(physics)
+
+biology.add_assignment("Bio exam #1")
+biology.add_assignment("Bio exam #2")
+physics.add_assignment("Physics homework #1")
+calculus.add_assignment("Calc homework #1")
+calculus.add_assignment("Calc quiz #1")
+
+biology.set_grade("Bio exam #1", john, 90)
+biology.set_grade("Bio exam #1", katie, 95)
+
+physics.set_grade("Physics homework #1", katie, 86)
+
+calculus.set_grade("Calc homework #1", john, 100)
+calculus.set_grade("Calc homework #1", katie, 98)
+calculus.set_grade("Calc quiz #1", john, 78)
+calculus.set_grade("Calc quiz #1", katie, 84)
+
+john.get_transcript()
+katie.get_transcript()
+```
```

Comment:
Could you break up this code a little bit? I feel like it'd be better to show one class, explain what the code is doing (for example, what's `self`?), and then show another code block showing how objects are instantiated, etc.